sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/DefaultSynchronousMetricStorage.java

```diff
@@ -27,20 +27,28 @@
  */
 public final class DefaultSynchronousMetricStorage<T, U extends ExemplarData>
     implements SynchronousMetricStorage {
+  private final RegisteredReader registeredReader;
   private final MetricDescriptor metricDescriptor;
+  private final AggregationTemporality aggregationTemporality;
   private final DeltaMetricStorage<T, U> deltaMetricStorage;
```

Comment:
If you're having ONE storage per-reader, you can actually remove the need for Delta vs. Temporal storage completely.

The assumption before was that DELTA + TEMPORAL => I can reconstitute state for any number of readers.

Given the direction this is going, you really want to GUT these, as they're only causing memory overhead (likely large GC cycles) in a more consistent manner in this change.

While i'm OOO travelling this week, would be happy to set up some time brainstorm/discuss alternatives here.

FWIW - Prior to my personal situation going sour in January, I was working on a refactor VERY similar to this one, just a bit more aggressive

Reply:
Nice observation! I've pushed a commit to merge the remaining minimal logic of DeltaMetricStorage into DefaultSynchronousMetricStorage